pyproject.toml

```diff
@@ -14,10 +14,23 @@ maintainers = [
 ]
 requires-python = ">=3.9"
 
-# All dependencies are optional; it depends on whether you are running
-# a client or server (or both) and what data structures you care about.
+# Dependencies common to both server and client; additional dependencies are
+# available depending on whether you are running client/server/both, and what
+# data structures you care about.
 # Use pip install tiled[all] to get everything.
-dependencies = []
+dependencies = [
+    "httpx >=0.20.0,!=0.23.1",
+    "json-merge-patch",
+    "jsonpatch",
+    "jsonschema",
+    "msgpack >=1.0.0",
+    "orjson",
+    "platformdirs",
+    "pydantic >=2, <3",
+    "pydantic-settings >=2, <3",
+    "pyyaml",
+    "typer",
```

Comment:
Is typer really required for client installs?

Reply:
We could follow the pattern that `httpx` uses and prompt the user to additionally install `typer` the first time they try to use the CLI. I am mixed on this. It can be annoying.

Reply:
Oh, I forgot about the registration features. 

I vote on the prompt.

Reply:
Oops... I vote `no` on the prompt to install typer.

Reply:
From a code readability standpoint, I'm opposed to conditional imports or imports in methods as far as is practicable. It's got 1 core and 2 optional dependencies, it's not a huge strain to have it by default.